src/trusted_ca_keys/trusted_ca_keys.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include "oslogin_utils.h"
+#include "trusted_ca_keys.h"
+
+using std::cout;
+using std::cerr;
+using std::endl;
+using std::flush;
+using std::string;
+using std::stringstream;
+using std::vector;
+using std::ofstream;
+
+using oslogin_utils::HttpGet;
+using oslogin_utils::ParseJsonToSuccess;
+using oslogin_utils::ParseJsonToEmail;
+using oslogin_utils::ParseJsonToSshKeys;
+using oslogin_utils::UrlEncode;
+using oslogin_utils::ParseJsonToCAKeys;
+
+ofstream
+CreateTmpKeysFile(const string& trusted_keys_filename, const string& trusted_keys_tmp_filename) {
+
+  ofstream trusted_keys_tmp_file(trusted_keys_tmp_filename.c_str());
+
+  if (!trusted_keys_tmp_file.is_open()) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Can't open '"
+            << trusted_keys_tmp_filename << "' for writing";
+    perror(err_msg.str().c_str());
+
+    exit(EXIT_FAILURE);
+  }
+
+  if (chown(trusted_keys_tmp_filename.c_str(), 0, 0) != 0) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Failed to set root:root file owner for "
+            << trusted_keys_tmp_filename;
+    perror(err_msg.str().c_str());
+    remove(trusted_keys_tmp_filename.c_str());
+
+    exit(EXIT_FAILURE);
+  }
+
+  if (chmod(trusted_keys_tmp_filename.c_str(),
+            S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH) != 0) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Failed to set 644 file permissions for "
+            << trusted_keys_tmp_filename;
+    perror(err_msg.str().c_str());
+    remove(trusted_keys_tmp_filename.c_str());
+
+    exit(EXIT_FAILURE);
+  }
+
+  return trusted_keys_tmp_file;
+}
+
+void
+WriteCAKeysTmpFile(ofstream& trusted_keys_tmp_file, const vector<string> ca_keys) {
```

Comment:
please, make it simple and move this loop to `main()` you have a single caller and it's not exactly doing anything special. Split it later if you need this logic elsewhere.

Reply:
Sorry, but I'd keep it as it is.
I see its simplicity in the readability.
Instead of adding a comment before the loop or even avoiding some comment, I prefer a function with a name that carries its purpose. I even wanted to move `rename` into a separate function for its readability, but it would have a similar name with that same only one instruction (so didn't make sense).

It is not only about code reusing or a number of callers.
I see a benefit in its readability over other less important here aspects.